// @datadog/browser-core/cjs/domain/trackingConsent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createTrackingConsentState = exports.TrackingConsent = void 0;
var experimentalFeatures_1 = require("../tools/experimentalFeatures");
var observable_1 = require("../tools/observable");
exports.TrackingConsent = {
    GRANTED: 'granted',
    NOT_GRANTED: 'not-granted',
};
function createTrackingConsentState(currentConsent) {
    var observable = new observable_1.Observable();
    return {
        tryToInit: function (trackingConsent) {
            if (!currentConsent) {
                currentConsent = trackingConsent;
            }
        },
        update: function (trackingConsent) {
            currentConsent = trackingConsent;
            observable.notify();
        },
        isGranted: function () {
            return (!(0, experimentalFeatures_1.isExperimentalFeatureEnabled)(experimentalFeatures_1.ExperimentalFeature.TRACKING_CONSENT) ||
                currentConsent === exports.TrackingConsent.GRANTED);
        },
        observable: observable,
    };
}
exports.createTrackingConsentState = createTrackingConsentState;
